std/container/rbtree.d

```diff
@@ -1714,10 +1714,13 @@ assert(equal(rbt[], [5]));
     Formats the RedBlackTree into a sink function. For more info see
     $(D std.format.formatValue)
     */
-    void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt) const {
-        sink("RedBlackTree(");
-        sink.formatValue(this[], fmt);
-        sink(")");
+    static if(is(typeof((){FormatSpec!(char) fmt; formatValue((const(char)[]) {}, ConstRange.init, fmt);})))
+    {
+        void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt) const {
```

Comment:
toString of Object doesn't have this form I don't think. But in any case we can save this for another time, the important thing is that this is fixing a regression where you can't use classes as element types